openid-4-verifiable-credential-issuance-1_0.md

```diff
@@ -1117,6 +1118,53 @@ Cache-Control: no-store
 }
 ```
 
+# Callback Endpoint {#callback_endpoint}
+
+This endpoint is used to receive notification from the Wallet whether credential has been successfully stored in the Wallet or not. Support for this endpoint is OPTIONAL.
+
+This endpoint can be used after the Credential Issuer has sent Credential Response or Batch Credential Response. It enables the Credential Issuer to take subsequent actions after issuance, depending on whether the credential has been accepted and successully stored by the Wallet, rejected by the Wallet, or errors and other unforeseen circumstances have occurred during the Wallet's processing.
+
+Communication with the Callback Endpoint MUST utilize TLS.
+
+## Callback from the Wallet
+
+The Wallet sends a callback to the Callback Endpoint by sending the following parameters in the entity-body of an HTTP POST request using the `application/json` media type.
+
+* `status`: REQUIRED. Status whether the credential issuance was successful or not. The value MUST be either `success` or `error`.
+* `error_description`: OPTIONAL. Human-readable ASCII [@!USASCII] text providing additional information, used to assist the Credential Issuer developer in understanding the error that occurred. Values for the `error_description`` parameter MUST NOT include characters outside the set %x20-21 / %x23-5B / %x5D-7E.
```

Comment:
WG has been looking for a parameter that the wallet includes in the callback so that the issuer can identify which credential this callback is about. based on our implementation what has been more relevant is which user session the callback is tied to. Suggestion would be to use a `state` parameter to be included in the callback. it is also probably the simplest when there is one callback for multiple credentials.

Reply:
"state" doesn't work very well for multiple credentials in the case that some of them are accepted and some are not. Also "state" is something used in the protocol exchange between client/wallet and AS so it's not really appropriate to try and use for something at the credential issuer/RS.

Reply:
If the credential has a unique identifier It then can be uniquely identifier

Jwt has the claim jti

I would suggest to pass also the credential type, because different types of credential may be processed by different microservices within the vci backend/architecture

Reply:
I think there is a confusion which state we are talking about. I was talking about issuer_state that the issuer passes in the credential_offer, not the state that the wallet sends in the authorization request.
That has also led me to think that issuer_state passed in the credential offer should be passed not in the authorization request, but the credential request.
I made changes to how issuer_state work - please take a look. I think with these new changes, issuer_state is the best suited for the purpose of callback.

Reply:
> If the credential has a unique identifier It then can be uniquely identified

I looked again and I don't think mdoc has equivalent of `jti`...